src/components/Alert/Alert.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+import { bem } from '../../utils';
+import { Button } from '../Buttons';
+import { Modal, ModalProps } from '../Modal';
+import styles from './Alert.scss';
+
+interface Props extends ModalProps {
+    /** Should the alert appear */
+    showAlert?: boolean;
+    /** The alert content */
+    children: ReactNode;
+    /** Dialog title */
+    title?: string;
+    /** Callback to be called on Accept button click (shown in both alert and confirm) */
+    onAccept: () => void;
+    /** Callback to be called on Cancel button click (only relevant for confirm dialog) */
+    onCancel: () => void;
+    /** Accept button's text */
+    acceptLabel: string;
+    /** Cancel button's text */
+    cancelLabel: string;
+}
+
+const { block, elem } = bem('Alert', styles);
+
+export const Alert: React.FC<Props> = (props) => {
+    const {
+        showAlert = false,
+        children,
+        title,
+        onAccept,
+        onCancel,
+        acceptLabel,
+        cancelLabel,
+        ...rest
+    } = props;
+
+    if ((!cancelLabel && !!onCancel) || (!!cancelLabel && !onCancel)) {
+        console.warn(
+            'It seems you are trying to render a confirm dialog. In that case make sure to pass both "onCancel" and "cancelLabel" props. Otherwise please remove both'
+        );
+    }
+
+    const isConfirm = !!cancelLabel;
```

Comment:
I'm a bit confused with this design. Here it seems like you infer from the presence (or not) of a cancel label / button, whether it's supposed to be an alert or confirm dialog. However, in the Storybook story I also see `type="[confirm |alert]"`. Which do you intend to use to make this distinction?
I think having a distinct `type` prop - even though more explicit / descriptive - may be superfluous because when `onCancel` is defined, it implicitly _has_ to be a confirm dialog.
Alternatively, if you want it to be more explicit, you could think of a separate Confirm component (which accepts onCancel / cancelLabel) that extends the Alert component (does not accept onCancel).

Reply:
I removed the example component from the story. Hopefully it is clearer now. (`type` was a prop on the example only, which implements the state for the alert to be shown or not)

Reply:
I like the idea of separating the Alert and Confirm dialog when Confirm extends the Alert component.